diofant/core/tests/test_cache.py

```diff
@@ -96,3 +98,20 @@ def test_nocache(clear_imports, monkeypatch):
     y = Symbol('y')
     result = s.subs(sqrt(x**2), y)
     assert result == 1/y
+
+    # problem from https://groups.google.com/forum/#!topic/sympy/LkTMQKC_BOw
+    # see commit c459d18
+    a = Symbol('a', positive=True)
+    f = exp(x*(-a - 1))
+    g = sinh(x)
+    r = f*g  # not raises
```

Comment:
@skirpichev Both problems are indeed not present in diofant.  They are still present in sympy 1.0.1_dev

Thank you very much.